module/input/SensorFilter/src/MotionModel.hpp

```diff
@@ -181,32 +170,30 @@ namespace module::input {
             return newState;
         }
 
-        Eigen::Matrix<Scalar, 3, 1> predict(const Eigen::Matrix<Scalar, size, 1>& state,
-                                            const MeasurementType::ACCELEROMETER&) {
-            // Extract our rotation quaternion
-            const Eigen::Matrix<Scalar, 3, 3> Rtw = StateVec(state).Rwt.inverse().toRotationMatrix();
-
-            // Make a world gravity vector and rotate it into torso space
-            // Where is world gravity with respect to robot orientation?
-            // Multiplying a matrix with (0, 0, G) is equivalent to taking the
-            // third column of the matrix and multiplying it by G
-            return Rtw.template rightCols<1>() * G;
+        Eigen::Matrix<Scalar, 3, 1> predict(const StateVec& state, const MeasurementType::ACCELEROMETER&) {
+
+            // Rotate world gravity vector into torso space using quaternion conjugation
+            // p' = q * p * q.conjugate()
+            // G' = Rtw * G * Rtw.conjugate()
+            //    = Rwt.conjugate() * G * Rwt.conjugate().conjugate()
+            //    = Rwt.conjugate() * G * Rwt
+            // G is a quaternion with real part zero and vector part (0, 0, G)
+            // The vector part of G' is the result of rotating G by Rtw
+            return (state.Rwt.conjugate() * Eigen::Quaternion<Scalar>(Scalar(0), Scalar(0), Scalar(0), Scalar(G))
+                    * state.Rwt)
+                .vec();
         }
 
-        Eigen::Matrix<Scalar, 3, 1> predict(const Eigen::Matrix<Scalar, size, 1>& state,
-                                            const MeasurementType::GYROSCOPE&) {
-            // Add predicted gyroscope bias to our predicted gyroscope
-            return StateVec(state).omegaTTt + StateVec(state).omegaTTt_bias;
+        Eigen::Matrix<Scalar, 3, 1> predict(const StateVec& state, const MeasurementType::GYROSCOPE&) {
+            return state.omegaTTt;
         }
 
-        Eigen::Matrix<Scalar, 3, 1> predict(const Eigen::Matrix<Scalar, size, 1>& state,
-                                            const MeasurementType::FLAT_FOOT_ODOMETRY&) {
-            return StateVec(state).rTWw;
+        Eigen::Matrix<Scalar, 3, 1> predict(const StateVec& state, const MeasurementType::FLAT_FOOT_ODOMETRY&) {
+            return state.rTWw;
         }
 
-        Eigen::Matrix<Scalar, 4, 1> predict(const Eigen::Matrix<Scalar, size, 1>& state,
-                                            const MeasurementType::FLAT_FOOT_ORIENTATION&) {
-            return StateVec(state).Rwt.coeffs();
+        Eigen::Matrix<Scalar, 4, 1> predict(const StateVec& state, const MeasurementType::FLAT_FOOT_ORIENTATION&) {
+            return {state.Rwt.w(), state.Rwt.x(), state.Rwt.y(), state.Rwt.z()};
```

Comment:
#528 clarifies the order they're stored and everything, and makes sure that this is fine
```suggestion
            return state.Rwt.coeffs();
```